packages/next/server/dev/hot-reloader.ts

```diff
@@ -396,8 +398,17 @@ export default class HotReloader {
     }
 
     const multiCompiler = webpack(configs)
+    const [clientCompiler, serverCompiler] = multiCompiler.compilers
+
+    // https://webpack.js.org/api/node/#custom-file-systems
+    if (isWebpack5) {
+      const filesystem = createFsFromVolume(new Volume())
+      // @ts-ignore In the future, we to use webpack5 types here
+      clientCompiler.outputFileSystem = filesystem
+      this.clientFileSystem = filesystem
+    }
```

Comment:
This `if` block is the only relevant change in this file.

The rest is using the `clientCompiler` and `serverCompiler` instead of `[0]` and `[1]`